pkg/controller/config/config_controller.go

```diff
@@ -257,6 +268,8 @@ func (r *ReconcileConfig) Reconcile(request reconcile.Request) (reconcile.Result
 		return reconcile.Result{}, fmt.Errorf("replaying data: %w", err)
 	}
 
+	r.processExcluder.Replace(instance.Spec.Match)
```

Comment:
It just occurred to me that we need to be able to handle the case where a namespace is excluded but resources have already been cached... those resources need to be culled somehow.

Reply:
How can we do this? I think we reset cache when sync changes today, right? Should we reset cache when `sync` exclusion gets updated?

Reply:
Ya, it's gross, but I think it's the best solution that's currently possible. It highlights the need to be able to query and prune cached data, which would allow us to get rid of full cache wipes altogether.

In the interim, we're relying on the fact that we're fail-open and that config changes to the watch list should be rare. Hopefully changes to the ignore set should be rare as well.

Reply:
Updated this to be similar to `watch`. Any change in excluded namespaces will reset the sync cache. If we handle `*` and `sync` separately from the rest, it gets more complex. Let me know what you think.

Reply:
+1 to using the blunt hammer that is less prone to bugs.